tests/ignite/metrics/test_metrics_lambda.py

```diff
@@ -345,8 +345,8 @@ def update_fn(engine, i):
 
         evaluator = Engine(update_fn)
 
-        precision = Precision(average=False, device=device)
-        recall = Recall(average=False, device=device)
+        precision = Precision(average=False, device=metric_device)
```

Comment:
What happens if metrics are different in expression terms ?

Reply:
That's a good question, I think it does not matter as lambda works on the results of compute which are float numbers... But worth to check

Reply:
I don't quite understand, what is the potential issue here?

Reply:
I think @sdesrozis wanted to know what would happen if we do (Sylvain, correct me I'm wrong)
```python
precision = Precision(average=False, device="cpu")
recall = Recall(average=False, device="gpu")
my_metric = precision * recall

my_metric.attach(engine, "my_metric")
```

Reply:
I understand now. Currently, this will fail when you call `compute` since you will be trying to multiply tensors on different devices. Is this behaviour ok? 

Otherwise, we can fail early by adding the following check to all the arithmetic operations in `Metric`:

```python
    def __mul__(self, other):
        from ignite.metrics.metrics_lambda import MetricsLambda
        if self._device != self._device:
            raise ValueError("Metrics combined with arithmetic must be on the same device. "
                             "Got self on {} and other on {}".format(self._device, other._device))
        return MetricsLambda(lambda x, y: x * y, self, other
```

Reply:
Well, I thought in most of the cases of our metrics `Metric.compute()` returns a scalar on cpu. That's why I thought there is no issue. I wonder if we would like to keep metric on other devices than cpu ?

Reply:
Yes, but sometimes (e.g. Precision and Recall when `average=False`) we return an entire tensor. I think it makes sense to keep the metrics on other devices sometimes (e.g. the GPU so the updates are quicker). 

In my opinion, the best solution is to move all the tensors to the CPU during the `compute` step for `MetricsLambda` (instead of throwing any errors). What do you think?

Reply:
I agree 👍

Reply:
Done in bcc3cbb.